LibGit2Sharp/Branch.cs

```diff
@@ -177,6 +177,16 @@ public virtual Remote Remote
                     return null;
                 }
 
+                // upstream git uses the "/" character to denote
+                // when a nick should not be resolved to a remote
+                // TODO:
+                // there's probably some pushspec stuff still to
+                // address here, but let me make this bugfix first
+                if (remoteName.Contains("/"))
```

Comment:
Wait I don't think that makes sense. We're just talking about the `remoteName` here not the branch name.